redox/processor_neworder.go

```diff
@@ -246,67 +246,68 @@ func (o *newOrderProcessor) handleCreateAccount(ctx context.Context, create Crea
-		return o.handleAccountCreationError(ctx, err, order, *match)
+		return false, o.handleAccountCreationError(ctx, err, order, *match)
 	}
 	createPatient.Mrn, err = GetMrnFromOrder(order)
 	if err != nil {
-		return o.handleAccountCreationError(ctx, err, order, *match)
+		return false, o.handleAccountCreationError(ctx, err, order, *match)
 	}
 
 	if createPatient.Email != nil {
 		if exists, err := o.emailExists(*createPatient.Email); err != nil {
 			o.logger.Errorw("unexpected error when checking for duplicate emails", "order", order.Meta, "error", err)
-			return err
+			return false, err
 		} else if exists {
 			err = fmt.Errorf("the email address is already in use")
-			return o.handleAccountCreationError(ctx, err, order, *match)
+			return false, o.handleAccountCreationError(ctx, err, order, *match)
 		}
 	}
 
 	createPatient.Tags, err = o.createTagsForPatient(ctx, order, *match)
 	if err != nil {
 		o.logger.Errorw("unexpected error when creating tags for patient", "order", order.Meta, "error", err)
-		return err
+		return false, err
 	}
 
 	resp, err := o.clinics.CreatePatientAccountWithResponse(ctx, *match.Clinic.Id, createPatient)
 	if err != nil {
 		// Retry in case of unexpected failure
 		o.logger.Errorw("unable to create patient account", "order", order.Meta, "error", err)
-		return err
+		return false, err
 	}
 	if (resp.StatusCode() != http.StatusOK && resp.StatusCode() != http.StatusConflict) || resp.JSON200 == nil {
 		// Retry in case of failure
 		o.logger.Errorw("unexpected response when creating patient account", "order", order.Meta, "statusCode", resp.StatusCode())
-		return err
+		return false, err
 	}
 
 	o.logger.Infow("patient account was successfully created", "order", order.Meta, "clinicId", match.Clinic.Id, "patientId", resp.JSON200.Id)
-	return o.handleAccountCreationSuccess(ctx, order, *match)
+	return true, o.handleAccountCreationSuccess(ctx, order, *match)
 }
 
-func (o *newOrderProcessor) handleCreateAccountAndEnableSummaryReports(ctx context.Context, createAndEnable CreateAccountEnableReports) error {
+func (o *newOrderProcessor) handleCreateAccountAndEnableSummaryReports(ctx context.Context, createAndEnable 	CreateAccountEnableReports) error {
 	// Checks if a matching account already exists without enabling reports
 	match, err := o.matchOrder(ctx, createAndEnable.GetMatchRequest(), createAndEnable.Order)
 	if err != nil {
 		return err
 	}
+
 	accountCreated := false
 	if match.Patients == nil || len(*match.Patients) == 0 {
-		create := CreateAccount{
-			DocumentId: createAndEnable.DocumentId,
-			Order:      createAndEnable.Order,
-		}
-		if err := o.handleCreateAccount(ctx, create); err != nil {
+		create := CreateAccount(createAndEnable)
+		if successfullyCreated, err := o.handleCreateAccount(ctx, create); err != nil {
 			return err
+		} else if !successfullyCreated {
+			// There was an error when creating the account, but a result was already sent
+			return nil
 		}
 		accountCreated = true
```

Comment:
Next time you're in this code, consider if `accountCreated` could be removed. I believe `successfullyCreated` could be used in its place?

Reply:
No, it can't as `successfullyCreated` is out of scope of the code that depends on `accountCreated`.